proposals/automating-releases.md

```diff
@@ -0,0 +1,185 @@
+- Manages code tags
+- Builds and pushes [binaries][goreleaser-build] and [images][goreleaser-docker]
+  - Applies to `operator-sdk`, `ansible-operator`, and `helm-operator`
+- Centralized configuration and a well-documented [configuration spec][goreleaser-config]
+- Handles the full release process with one click
+
+`goreleaser` configuration is CI-agnostic, so if we choose to switch CI systems in the future
+we can bring our configuration with us verbatim.
+
+Additionally it supports generating release notes based on commits, or from a
+[custom source][goreleaser-custom-changelog]; we currently use a custom changelog generator
+which we can continue using.
+
+### Risks and Mitigations
+
+- Contributors need to learn a new configuration format. Luckily the goreleaser
+[config file format][goreleaser-config] is straightforward and has great documentation.
+- Release tool development is controlled by another organization, which can lead to undesirable bugs/changes.
+This is a manageable risk because the project is open source, to which we can contribute.
+
+## Design Details
+
+### Test Plan
+
+1. Create a goreleaser configuration that creates the exact same release artifacts the current process does.
+1. Push configuration to a test repository.
+1. Run a test release.
+1. Ensure:
+  1. Tag is present remotely.
+  1. Release notes/binaries are published as expected.
+  1. Images are pullable and run on the desired architectures.
+
+### Graduation Criteria
+
+Once test plan has been executed and passed, the goreleaser configuration can be pushed to the
+operator-sdk repo and unused release scripts/code removed.
+
+#### Examples
+
+TBD
+
+##### Dev Preview -> Tech Preview
+
+- Ability to utilize the enhancement end to end
+- End user documentation, relative API stability
+- Sufficient test coverage
+- Gather feedback from users rather than just developers
+
+##### Tech Preview -> GA
+
+- More testing (upgrade, downgrade, scale)
+- Sufficient time for feedback
+- Available by default
+
+**For non-optional features moving to GA, the graduation criteria must include
+end to end tests.**
+
+##### Removing a deprecated feature
+
+N/A
+
+### Upgrade / Downgrade Strategy
+
+N/A
+
+### Version Skew Strategy
+
+The `goreleaser` binary will be updated whenever a new release containing a bug fix we want is available.
+
+## Implementation History
+
+Current release [script][sdk-release-script] and [documentation][sdk-release-doc].
+
+## Drawbacks
+
+See [risks and mitigations](#risks-and-mitigations).
+
+## Alternatives
+
+- Combining native Github actions like [create-release](https://github.com/actions/create-release).
+  - Still requires glue code and multiple actions have different configs that are not documented as well.
+- A traditional build system like [Bazel][bazel]. While this could work, it is quite heavy-handed for
+what we need and would likely involve writing glue code.
```

Comment:
-1 to bazel. I'm okay for it to be listed as an alternative, but I agree it's heavy-handed.